recipe/patches/0001-Locate-zoneinfo-based-on-install-prefix.patch

```diff
@@ -0,0 +1,255 @@
+From 39babe9340b29f82e6a70118ac6b11fd0f04808a Mon Sep 17 00:00:00 2001
+From: Austin Morton <austin.morton@aquatic.com>
+Date: Tue, 20 Aug 2024 14:09:50 +0000
+Subject: [PATCH] Locate zoneinfo based on install prefix
+
+Patch based on https://github.com/conda-forge/ctng-compilers-feedstock/blob/1484ff49f71cac78580cbf6c4aab350b2415cf49/recipe/patches/0003-patch-zoneinfo_dir_override-to-point-to-our-tzdata.patch
+---
+ CMakeLists.txt    |   4 +-
+ include/date/tz.h |   6 ---
+ src/tz.cpp        | 127 +++++++++++++++++++++++++++++-----------------
+ 3 files changed, 82 insertions(+), 55 deletions(-)
+
+diff --git a/CMakeLists.txt b/CMakeLists.txt
+index 012512a..1a0edab 100644
+--- a/CMakeLists.txt
++++ b/CMakeLists.txt
+@@ -116,7 +116,7 @@ if( BUILD_TZ_LIB )
+           src/ios.mm )
+     endif()
+     add_library( date::date-tz ALIAS date-tz )
+-    target_link_libraries( date-tz PUBLIC date )
++    target_link_libraries( date-tz PUBLIC date ${CMAKE_DL_LIBS} )
+     target_include_directories( date-tz PUBLIC
+         $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>
+         $<INSTALL_INTERFACE:include> )
+@@ -127,7 +127,7 @@ if( BUILD_TZ_LIB )
+       target_compile_definitions( date-tz PRIVATE AUTO_DOWNLOAD=1 HAS_REMOTE_API=1 )
+     endif()
+ 
+-    if ( USE_SYSTEM_TZ_DB AND NOT WIN32 AND NOT MANUAL_TZ_DB )
++    if ( USE_SYSTEM_TZ_DB AND NOT MANUAL_TZ_DB )
+       target_compile_definitions( date-tz PRIVATE INSTALL=. PUBLIC USE_OS_TZDB=1 )
+     else()
+       target_compile_definitions( date-tz PUBLIC USE_OS_TZDB=0 )
+diff --git a/include/date/tz.h b/include/date/tz.h
+index 4921068..03c0042 100644
+--- a/include/date/tz.h
++++ b/include/date/tz.h
+@@ -82,12 +82,6 @@ static_assert(HAS_REMOTE_API == 0 ? AUTO_DOWNLOAD == 0 : true,
+ #  define USE_SHELL_API 1
+ #endif
+ 
+-#if USE_OS_TZDB
+-#  ifdef _WIN32
+-#    error "USE_OS_TZDB can not be used on Windows"
+-#  endif
+-#endif
+-
+ #ifndef HAS_DEDUCTION_GUIDES
+ #  if __cplusplus >= 201703
+ #    define HAS_DEDUCTION_GUIDES 1
+diff --git a/src/tz.cpp b/src/tz.cpp
+index 26babbd..5de4f05 100644
+--- a/src/tz.cpp
++++ b/src/tz.cpp
+@@ -128,6 +128,7 @@
+ #  endif
+ 
+ #  include <io.h> // _unlink etc.
++#  include <intrin.h>
+ 
+ #  if defined(__clang__)
+     struct IUnknown;    // fix for issue with static_cast<> in objbase.h
+@@ -154,6 +155,7 @@
+ #    include <sys/wait.h>
+ #    include <sys/types.h>
+ #  endif //!USE_SHELL_API
++#  include <dlfcn.h>
+ #endif  // !_WIN32
+ 
```

Comment:
One of my previous comment might not have been clear enough, I meant that we could use:
```suggestion
 
#include <string>
```

Reply:
for what? `<string>` is already included in this file.

Reply:
I was thinking of `<chrono>` potentially not including it in the future.

Anyway, that's a nitpick.

Reply:
right - I mean this file already has `#include <string>` - the comment about `<chrono>` was from the upstream patch and not relevant here.